app/models/miq_alert.rb

```diff
@@ -545,17 +546,27 @@ def evaluate_in_automate(target, inputs = {})
   end
 
   def evaluate_internal(target, _inputs = {})
-    method = "evaluate_method_#{expression[:eval_method]}"
+    if target.class.name == 'MiddlewareServer'
```

Comment:
Yes, alerting should be extended to other middleware models (datasources, messaging, etc) and probably that will drive future enhancements on the provider side (i.e. extracting more common code into mixin or base classes).
That is part of the future work.